modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/DLOpenerOneDriveManager.java

```diff
@@ -138,6 +139,19 @@ public void deleteFile(long userId, FileEntry fileEntry)
 					fileEntry);
 		}
 		catch (GraphServiceException graphServiceException) {
+			GraphServicePortalException graphServicePortalException =
+				GraphServiceExceptionPortalExceptionMapper.map(
+					graphServiceException);
+
+			if (graphServicePortalException instanceof
+					GraphServicePortalException.ItemNotFound) {
+
+				_dlOpenerFileEntryReferenceLocalService.
+					deleteDLOpenerFileEntryReference(
+						DLOpenerOneDriveConstants.ONE_DRIVE_REFERENCE_TYPE,
+						fileEntry);
+			}
+
 			throw GraphServiceExceptionPortalExceptionMapper.map(
 				graphServiceException);
```

Comment:
Why are you calling the `map` method twice? Shouldn't you simply `throw graphServicePortalException`?

Reply:
You're absolutely right. I'll fix that.